### PR TITLE
[py] Updated Handling for DetachedShadowRoot Exception

### PR DESCRIPTION
**Pull request by shbenzer**
### **User description**
Added DetachedShadowRootException to python error handling

### Description
Added DetachedShadowRootException
Added parameterized test

### Motivation and Context
#13580 

### Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### Checklist

- [x] I have read the [contributing](https://github.com/SeleniumHQ/selenium/blob/trunk/CONTRIBUTING.md) document.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

___

### **PR Type**
Bug fix, Tests

___

### **Description**
- Added a new exception `DetachedShadowRootException` to handle cases where a shadow root is no longer attached to the DOM.
- Updated the error handler to map a specific error code to `DetachedShadowRootException`.
- Enhanced unit tests to include a test case for `DetachedShadowRootException`.

___

PRDescriptionHeader.CHANGES_WALKTHROUGH
<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Enhancement</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>__init__.py</strong><dd><code>Add DetachedShadowRootException to common module</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

py/selenium/common/__init__.py

<li>Added import for <code>DetachedShadowRootException</code>.<br> <li> Included <code>DetachedShadowRootException</code> in the <code>__all__</code> list.<br>

</details>

  </td>
  <td><a href="https://github.com/SeleniumHQ/selenium/pull/14677/files#diff-17868cc5f55d0898c224b4f2f8014a8092fcb9ecd382776a9b873142e9f1990a">+2/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>exceptions.py</strong><dd><code>Define DetachedShadowRootException class</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

py/selenium/common/exceptions.py

<li>Introduced <code>DetachedShadowRootException</code> class.<br> <li> Documented the purpose of the exception.<br>

</details>

  </td>
  <td><a href="https://github.com/SeleniumHQ/selenium/pull/14677/files#diff-1b2829f22ba2d98a53d9f003d5f04dbfe70ff4447409c32dcc475fca0cdec754">+3/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>errorhandler.py</strong><dd><code>Map DetachedShadowRootException in error handler</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

py/selenium/webdriver/remote/errorhandler.py

<li>Imported <code>DetachedShadowRootException</code>.<br> <li> Mapped error code to <code>DetachedShadowRootException</code>.<br>

</details>

  </td>
  <td><a href="https://github.com/SeleniumHQ/selenium/pull/14677/files#diff-3a2cce6f647dc882c5111a63c86a273646543e70b1ba3538181aee04a9385d78">+3/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    
</table></td></tr><tr><td><strong>Tests</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>error_handler_tests.py</strong><dd><code>Add test case for DetachedShadowRootException</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

py/test/unit/selenium/webdriver/remote/error_handler_tests.py

<li>Added test for <code>DetachedShadowRootException</code>.<br> <li> Used parameterized test for error code.<br>

</details>

  </td>
  <td><a href="https://github.com/SeleniumHQ/selenium/pull/14677/files#diff-16a7f3559108027005862f77d542ca48d5dc188f7325ca4df1b73984f7de86c5">+5/-1</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    
</table></td></tr></tr></tbody></table>

___

> 💡 **PR-Agent usage**: Comment `/help "your question"` on any pull request to receive relevant information

### CODE REVIEW

File: py/selenium/common/exceptions.py [hunk 1]
```diff
@@ -285,3 +285,6 @@ def __init__(
         with_support = f"{msg}; {SUPPORT_MSG} {ERROR_URL}/driver_location"
 
         super().__init__(with_support, screen, stacktrace)
+
+class DetachedShadowRootException(WebDriverException):
+    """Raised when referenced shadow root is no longer attached to the DOM"""
```
Comment (Delta456) [inline py/selenium/common/exceptions.py]: add new line here

File: py/selenium/webdriver/remote/errorhandler.py
Comment (Delta456) [inline py/selenium/webdriver/remote/errorhandler.py:1]: Please run `scripts/format.sh` to format the files

File: py/selenium/common/exceptions.py (hunk 1 above)
Comment (shbenzer) [reply py/selenium/common/exceptions.py]: added

File: py/selenium/webdriver/remote/errorhandler.py
Comment (shbenzer) [reply py/selenium/webdriver/remote/errorhandler.py:1]: Done

**Comment by shbenzer** [comment]
@Delta456 Should be all good now

**Comment by AutomatedTester** [comment]
Please change the commit message to reflect the change. The exception already exists and hasn't not been added by this PR

**Comment by shbenzer** [comment]
> Please change the commit message to reflect the change. The exception already exists and hasn't not been added by this PR

Is this better?

**Comment by shbenzer** [comment]
> Please change the commit message to reflect the change. The exception already exists and hasn't not been added by this PR

@AutomatedTester is this better?

**Review by AutomatedTester — APPROVED** [review]

**Comment by shbenzer** [comment]
It was a missing period... My bad. Should be ready for merging now @AutomatedTester

**Comment by shbenzer** [comment]
Failed WebDriverWait test has nothing to do with pr

**Comment by shbenzer** [comment]
Failed WebDriverWait test has nothing to do with pr @AutomatedTester 

> py/test/selenium/webdriver/common/webdriverwait_tests.py::test_should_wait_until_all_visible_elements_are_found_when_searching_for_many[firefox] FAILED [ 24%]
> _ test_should_wait_until_all_visible_elements_are_found_when_searching_for_many[firefox] _
> driver = <selenium.webdriver.firefox.webdriver.WebDriver (session="9bda52a0-a27c-453f-b75e-7b683abe6c1a")>
> pages = <conftest.pages.<locals>.Pages object at 0x7f3a1188a9a0>
>     def test_should_wait_until_all_visible_elements_are_found_when_searching_for_many(driver, pages):
>         pages.load("hidden_partially.html")
>         add_visible = driver.find_element(By.ID, "addVisible")
>         add_visible.click()
>         add_visible.click()
>         elements = WebDriverWait(driver, 3).until(EC.visibility_of_all_elements_located((By.CLASS_NAME, "redbox")))
>        assert len(elements) == 2
> E       assert 1 == 2
> E        +  where 1 = len([<selenium.webdriver.remote.webelement.WebElement (session="9bda52a0-a27c-453f-b75e-7b683abe6c1a", element="aa6c4e85-d4a9-4c8d-baf7-f5fb0c184eea")>])
> py/test/selenium/webdriver/common/webdriverwait_tests.py:112: AssertionError

**Comment by shbenzer** [comment]
@VietND96 these new failures have nothing to do with this code.

**Pull request merged**